src/internal_modules/roc_fec/writer.cpp

```diff
@@ -109,6 +114,7 @@ status::StatusCode Writer::write(const packet::PacketPtr& pp) {
     }
 
     if (cur_packet_ == 0) {
+        update_block_duration_(pp);
```

Comment:
I suggest to call this from begin_block_ or end_block_ - so that everything called on block boundary would be in same place.